rust/main/chains/hyperlane-radix/src/provider/radix.rs

```diff
@@ -0,0 +1,723 @@
+        let arguments = manifest_encode(argument).map_err(HyperlaneRadixError::from)?;
+
+        self.call_method(component, method, state_version, vec![arguments])
+            .await
+    }
+
+    /// Returns the latest ledger state of the chain
+    pub async fn get_status(&self, reorg: &ReorgPeriod) -> ChainResult<LedgerState> {
+        let status = self.gateway_status().await?;
+        let mut state = status.ledger_state;
+
+        let offset = match reorg {
+            ReorgPeriod::None => 0,
+            ReorgPeriod::Blocks(blocks) => blocks.get(),
+            ReorgPeriod::Tag(_) => {
+                return Err(HyperlaneRadixError::Other(
+                    "radix only supports blocks as reorg periods".to_owned(),
+                )
+                .into())
+            }
+        };
+        state.state_version = state.state_version.saturating_sub(offset as i64);
+        Ok(state)
+    }
+
+    fn filter_parsed_logs<T: ScryptoEvent>(
+        contract: &str,
+        txs: Vec<CommittedTransactionInfo>,
+        parse: fn(ProgrammaticScryptoSborValue) -> ChainResult<T>,
+    ) -> ChainResult<Vec<(T, LogMeta)>> {
+        let mut events = vec![];
+
+        for tx in txs {
+            let Some(receipt) = tx.receipt else {
+                return Err(HyperlaneRadixError::ParsingError("receipt".to_owned()).into());
+            };
+            let Some(hash) = tx.intent_hash else {
+                return Err(HyperlaneRadixError::ParsingError(
+                    "failed to parse intent hash".to_owned(),
+                )
+                .into());
+            };
+            let (_, hash) = bech32::decode(&hash).map_err(HyperlaneRadixError::from)?;
+            let hash = H256::from_slice(&hash);
+            let Some(raw_events) = receipt.events else {
+                return Err(
+                    HyperlaneRadixError::ParsingError("events no present".to_owned()).into(),
+                );
+            };
+
+            for (event_index, event) in raw_events.iter().enumerate() {
+                // make sure to return only events that match the params
+                if event.name != T::EVENT_NAME {
+                    continue;
+                }
+                let emitter: EventEmitterIdentifier = serde_json::from_value(event.emitter.clone())
+                    .map_err(HyperlaneRadixError::from)?;
+                match emitter {
+                    EventEmitterIdentifier::Method(method)
+                        if method.entity.entity_address == contract =>
+                    {
+                        let address = decode_bech32(&method.entity.entity_address)?;
+
+                        // Pad address to 32 bytes with zeros
+                        let mut padded_address = [0u8; 32];
+                        let len = std::cmp::min(address.len(), 32);
+                        padded_address[32 - len..].copy_from_slice(&address[..len]);
+                        let address: H256 = padded_address.into();
```

Comment:
I wonder if it is the same as `Bytes32` address and if we can use `address_to_h256` here?